docs/tutorial/index.rst

```diff
@@ -1,4 +1,147 @@
+
+    >>> schedule = sorted(schedule, key=lambda item: item.slot.starts_at)
+    >>> for item in schedule:
+    ...     print(f"{item.event.name} at {item.slot.starts_at} in {item.slot.room.name}")
+    Talk 3 at 15-Sep-2016 09:30 in Small
+    Talk 11 at 15-Sep-2016 09:30 in Big
+    Talk 4 at 15-Sep-2016 10:00 in Small
+    Talk 10 at 15-Sep-2016 10:00 in Big
+    Talk 5 at 15-Sep-2016 12:30 in Big
+    Talk 9 at 15-Sep-2016 12:30 in Small
+    Talk 6 at 15-Sep-2016 13:00 in Big
+    Talk 8 at 15-Sep-2016 13:00 in Small
+    Talk 1 at 16-Sep-2016 09:30 in Big
+    Workshop 2 at 16-Sep-2016 09:30 in Small
+    Talk 2 at 16-Sep-2016 10:00 in Big
+    Talk 7 at 16-Sep-2016 12:30 in Big
+    City tour at 16-Sep-2016 12:30 in outside
+    Talk 12 at 16-Sep-2016 13:00 in Big
+    Workshop 1 at 16-Sep-2016 13:00 in Small
+    Boardgames at 16-Sep-2016 13:00 in outside
+
+
+
+We see that all the events are scheduled in appropriate rooms (as indicated by
+the unavailability attribute for the events). Also we have that :code:`Talk 1`
+doesn't class with :code:`Workshop 1`.
+Similarly, the :code:`Boardgame` does not clash with :code:`Workshop 2`.
+
+You will also note that in any given session, talks share at least one tag. This
+is another constraint of the model, if you find that your schedule has no
+solutions you can adjust it by recategorising your talks (or giving them all a
+single category).
+
+The data we input in to the model included information about demand for a talk,
+this could be approximated from previous popularity for a talk. However, the
+scheduler has put this talk in the small room (which has capacity 50). We can
+include an objective function in to our scheduler to aim to minimise the
+difference between room capacity and demand::
+
+    >>> from conference_scheduler.lp_problem import objective_functions
+    >>> func = objective_functions.capacity_demand_difference
+    >>> schedule = scheduler.schedule(events, slots, objective_function=func)
+
+    >>> schedule = sorted(schedule, key=lambda item: item.slot.starts_at)
+    >>> for item in schedule:
+    ...     print(f"{item.event.name} at {item.slot.starts_at} in {item.slot.room.name}")
+    Talk 1 at 15-Sep-2016 09:30 in Big
+    Talk 12 at 15-Sep-2016 09:30 in Small
+    Talk 3 at 15-Sep-2016 10:00 in Big
+    Talk 11 at 15-Sep-2016 10:00 in Small
+    Talk 4 at 15-Sep-2016 12:30 in Small
+    Talk 10 at 15-Sep-2016 12:30 in Big
+    Talk 2 at 15-Sep-2016 13:00 in Small
+    Talk 9 at 15-Sep-2016 13:00 in Big
+    Talk 7 at 16-Sep-2016 09:30 in Big
+    Workshop 2 at 16-Sep-2016 09:30 in Small
+    Talk 6 at 16-Sep-2016 10:00 in Big
+    Talk 8 at 16-Sep-2016 12:30 in Big
+    City tour at 16-Sep-2016 12:30 in outside
+    Talk 5 at 16-Sep-2016 13:00 in Big
+    Workshop 1 at 16-Sep-2016 13:00 in Small
+    Boardgames at 16-Sep-2016 13:00 in outside
+
+
+We see that :code:`Talk 3` has moved to the bigger room but that all other
+constraints still hold.
```

Comment:
> t seems odd to me that trying to match talks to room sizes isn't the default behaviour? Are they randomly assigned otherwise? What’s the default behaviour?

The main reason for this is because we want to be able to have multiple objective functions so by default the linear programming algorithm is just finding "a" solution (of which there are potentially many).

> As a side note, maybe this would be a good opportunity to show how to adjust a schedule? "We've had a surge in demand for Talk X, let's adjust the demand, and re-run the scheduler to tweak the schedule. Look, it's barely changed."

Good idea but I think I'd like to leave those extra things for now: mainly putting this tutorial together as a starting point for doc driven development so that we can make sure we're thinking of things in the right way. I expect we'll add more things like that :)